base/src/main/scala/co/uproot/abandon/Config.scala

```diff
@@ -27,16 +28,56 @@ object SettingsHelper {
     }
   }
 
+  def createTxnFilter(key: String, value: String): TransactionFilter = {
+    def makeDate(date: String) = {
+      val jDate = java.time.LocalDate.parse(date,
+          java.time.format.DateTimeFormatter.ISO_DATE)
+      Date(jDate.getYear, jDate.getMonthValue, jDate.getDayOfMonth)
+    }
+
+    (key, value) match {
+      case (key, value) if (key == "begin") => {
+        BeginDateTxnFilter(makeDate(value))
+      }
+      case (key, value) if (key == "end") => {
+        EndDateTxnFilter(makeDate(value))
+      }
+      case (key, value) if (key == "payee") => {
+        PayeeTxnFilter(value)
+      }
+      case (key, value) if (key == "account") => {
+        AccountTxnFilter(value)
+      }
+      case (key, value) if (key == "annotation") => {
+        AnnotationTxnFilter(value)
+      }
+      case _ => {
+        // TODO fix this
+        throw new RuntimeException("unknown filter")
+      }
+    }
+  }
+
   def getCompleteSettings(args: Seq[String]) = {
     val cliConf = new AbandonCLIConf(args)
+
+    val txnFilters =
+      if (cliConf.filters.isEmpty) {
+        None
```

Comment:
On second thoughts, the current code is better because you can issue warnings about filters more easily.